README.md

```diff
@@ -1 +1,679 @@
+
+For instance, if we are interested in the fact that we ended up with `None` in our composition, we can change the code like so:
+
+```fsharp
+let printValue (value: string option) =
+    match value with
+        | Some v -> v |> printfn "%s"
+        | _ -> printfn "No value available"
+
+success_vars
+    |> readLines
+    |> Seq.map tokenize
+    |> Seq.choose each
+    |> Seq.filter (keyToUpper >> keyContains "ELEMENT")
+    |> Seq.sort 
+    |> Seq.map (values >> toInt)
+    |> Seq.fold sum (Some 0)
+    >>= format
+    |> printValue
+```
+
+The result is now evaluated as:
+
+```sh
+❯ dotnet fsi pipes.fsx
+No value available
+❯ 
+```
+
+More importantly, though, when wrapped in a bind, our function will only be called given a valid, existing value. We can trust our computational flow to handle the data; thus, we can focus on the logic.
+
+And this is where things start to fall apart in javascript.
+
+Remember our previous example in javascript, and change the code as we did in f#.
+
+```js
+let test_data = `
+ELEMENT01=3
+ELEMENT04=4
+discarded:incorrect
+ELEMENT03=1
+foo=4
+element06=not an integer
+ELEMENT07=12
+bar=3
+ELEMENT02=12
+`
+test_data
+    |> readLines(^)
+    |> tokenize(^)
+    |> ^.filter(e => e |> keyToUpper(^) |> keyContains("ELEMENT", ^))
+    |> ^.sort(sort)
+    |> ^.reduce(sumValues, 0)
+    |> format(^)
+    |> console.log(^)
+```
+
+When we execute this, the result will be:
+
+```sh
+Done
+```
+
+This output is because our computational flow is broken starting from the `reduce` statement. In the `sumValues` function, `element06` will become `NaN`, and all sums after that will be `NaN`. In our format function, as we didn't account for `NaN`, we will return the result of the `else` statement, and the `console.log` is without an option to decide if the result is valid and thus should be printed.
+
+Let's look at the main differences:
+
+- In f#, once we have encountered our first `None`, the `add` is no longer called, as `bind` prevents that. So in javascript, after the current value is `NaN`, addition continues for the other numbers. This is the great advantage of optional types that we have seen in the second chapter.
+- In f#, the format function would be called only for valid results from the sum, as we added the binding. In javascript, this is always done.
+- As a result, we can either use the binding again for the final print in f# or branch it - based on the outcome. Unfortunately, the javascript code is entirely unable to decide the result at that point.
+
+To achieve the same result, we will first have to invent optional types and binding for javascript. Without it, we cannot create the laws required to prove that our flow will succeed or fail. And thus, like in our example, it will error in a way that will be very hard to trace, debug and understand in a more complex application - leading to the exact opposite of what functional paradigms usually provide.
```

Comment:
```suggestion
To achieve the same result, we will first have to invent optional types and binding for javascript. Without it, we cannot create the laws required to prove that our flow will succeed or fail. And thus, like in our example, it will error in a way that will be very hard to trace, debug, and understand in a more complex application - leading to the exact opposite of what functional paradigms usually provide.
```